sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/count/CountSamples.groovy

```diff
@@ -0,0 +1,66 @@
+package life.qbic.business.samples.count
+
+import life.qbic.business.DataSourceException
+import life.qbic.datamodel.samples.Status
+
+/**
+ * <b>Count samples</b>
+ *
+ * <p>This use case counts samples of a project and the subset of samples that have been received at the lab.</p>
+ *
+ * @since 1.0.0
+ */
+class CountSamples implements CountSamplesInput{
+  private final CountSamplesDataSource dataSource
+  private final CountSamplesOutput output
+  // used to be able to count samples for having completed a status when they are in a further step
+  private final List<Status> statusesInOrder = [Status.METADATA_REGISTERED, Status.SAMPLE_RECEIVED,
+    Status.SAMPLE_QC_FAIL, Status.FAILED_QC, Status.SAMPLE_QC_PASS,
+    Status.LIBRARY_PREP_FINISHED, Status.SEQUENCING, Status.SEQUENCING_COMPLETE,
+    Status.DATA_AVAILABLE, Status.DATA_AT_QBIC]
```

Comment:
I think we are not using these statuses, maybe @KochTobi can elaborate on that?

```suggestion
  // used to be able to count samples for having completed a status when they are in a further step
  private final List<Status> statusesInOrder = [Status.METADATA_REGISTERED, Status.SAMPLE_RECEIVED,
    Status.SAMPLE_QC_FAIL, Status.SAMPLE_QC_PASS,
    Status.LIBRARY_PREP_FINISHED, Status.DATA_AVAILABLE]
```
I think its just these statuses

Reply:
You are correct @jenniferboedker 
```
final static List<Status> SAMPLE_STATUSES = [Status.SAMPLE_RECEIVED, Status.SAMPLE_QC_PASS, Status.SAMPLE_QC_FAIL, Status.LIBRARY_PREP_FINISHED]
```

Reply:
thx